From mdharm@multivac.one-eyed-alien.net Wed Oct 10 10:56:39 2007
From: Richard Sharpe <rsharpe@richardsharpe.com>
Date: Wed, 10 Oct 2007 10:56:28 -0700
Subject: USB: allow usbstorage to have LUNS greater than 2Tb
To: USB Storage List <usb-storage@lists.one-eyed-alien.net>, Greg KH <greg@kroah.com>
Message-ID: <20071010175628.GA13290@one-eyed-alien.net>
Content-Disposition: inline

From: Richard Sharpe <rsharpe@richardsharpe.com>

Attached is a very small patch (several comment lines) and a one-line
coded change) that allows for USB storage devices that are larger than
2TB.

At the company where I work we need such support, and one of my
co-workers, Jane Liu, pointed out that SCSI low-layer drivers need to
specify what size CDBs they accept. After looking through the code it
became obvious that the current USB Storage code accepted the default of
12-byte CDBs, so I changed it to accept 16-byte CDBs. This allows our
device to work.

Signed-off-by: Richard Sharpe <rsharpe@richardsharpe.com>
Signed-off-by: Matthew Dharm <mdharm-usb@one-eyed-alien.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/storage/usb.c |    4 ++++
 1 file changed, 4 insertions(+)

--- a/drivers/usb/storage/usb.c
+++ b/drivers/usb/storage/usb.c
@@ -960,6 +960,10 @@ static int storage_probe(struct usb_inte
 		return -ENOMEM;
 	}
 
+	/*
+	 * Allow 16-byte CDBs and thus > 2TB
+	 */
+	host->max_cmd_len = 16;
 	us = host_to_us(host);
 	memset(us, 0, sizeof(struct us_data));
 	mutex_init(&(us->dev_mutex));
